document/CoreFormat.mdk

```diff
@@ -2225,3 +2260,112 @@ of the resulting UTF-8 byte sequence.
 
 * TODO: Relationship between original file encoding, offset, and UTF-8 byte sequence.
 * TODO: Explain responsibility of viewer/editor to match encoding.
+
+## `notification` object { #notification }
+
+### General { #notification-General }
+
+A `notification` object describes a condition encountered in the course of running an analysis tool
+which is relevant to the operation of the tool itself,
+as opposed to being relevant to a file being analyzed by the tool.
+Conditions relevant to files being analyzed by a tool are represented by `result` objects (&sect;[#result]).
+
+### `id` property { #notification-id }
+
+A `notification` object may contain a property named `id` whose value is a string containing
+the stable, opaque identifier for the condition that was encountered.
+
+### `ruleId` property { #notification-ruleId }
+
+If the condition described by the `notification` object is relevant to a particular analysis rule,
+the `notification` property should contain a property named `ruleId` whose value is the stable,
+unique identifier of the rule (&sect;[#rule-id]).
+
+### `message` property { #notification-message }
+
+A `notification` object shall contain a property named `message` whose value is a string that describes the condition that was encountered.
+
+### `level` property { #notification-level }
+
+A `notification` object may contain a property named `level` whose value is one of a fixed set of strings
+that specify the severity level of the notification.
+
+If present, the `level` property shall have one of the following values, with the specified meanings:
+
+* `error`: A serious problem was found.
+
+* `warning`: A problem that is not considered to be serious was found.
+
+* `note`: The notification is purely informational.
+
+If the `level` property is absent, it shall be considered equivalent to the value `warning`.
+
+### `time` property { #notification-time }
+
+A `notification` object may contain a property named `time` whose value is a string
+specifying the date and time at which the analysis tool generated the notification.
+The string shall be in the format specified by (&sect;[#fileFormat-dateTime]).
+
+### `exception` property { #notification-exception }
+
+If the issue is a result of a runtime exception, the `notification` object may contain a property named `exception`
```

Comment:
Fixed.

---
In reply to: [61360057](https://github.com/sarif-standard/sarif-spec/pull/172#discussion_r61360057) [](ancestors = 61360057)